tests/rptest/tests/full_disk_test.py

```diff
@@ -642,5 +650,10 @@ def target_size_reached():
                 )
             return below
 
-        # give it plenty of time. on debug it is hella slow
-        wait_until(target_size_reached, timeout_sec=120, backoff_sec=5)
+        if cleanup_policy == TopicSpec.CLEANUP_COMPACT:
+            # For `cleanup.policy=compat` we don't expect any removal. Wait for timeout.
+            with expect_exception(TimeoutError, bool):
+                wait_until(target_size_reached, timeout_sec=120, backoff_sec=5)
```

Comment:
nit: maybe have this wait be small factor of the compaction interval instead? E.g. set `log_compaction_interval_ms` to some 5s and wait for 5 * 5s? It'd feel a bit less arbitrary than 120s, though I don't feel strongly about it other than it seems like waiting 2 minutes for this test is a bit wasteful